zone/spell_effects.cpp

```diff
@@ -7415,3 +7415,30 @@ void Client::BreakFeignDeathWhenCastOn(bool IsResisted)
 		MessageString(Chat::SpellFailure,FD_CAST_ON);
 	}
 }
+
+bool Mob::HarmonySpellLevelCheck(int32 spell_id, int target_id, Mob *target)
```

Comment:
I love that you made this a function to help aggregate this check

The thing that I would suggest to simplify this is to just pass the mob target entity in this where you're passing in `target_id`

Eg. If you need to fetch the entity out of band; do it in the place where this function is being called and pass the mob entity into the function 

```cpp
auto target = entity_list.GetMobID(target_id);

// ...
if (!HarmonySpellLevelCheck(spell_id, target)) {

```